python_ta/debug/snapshot.py

```diff
@@ -14,6 +33,12 @@ def snapshot():
     local_vars = []
     frame = inspect.currentframe().f_back
 
+    # Handling possible errors
+    if frame is None:
+        raise RuntimeError("Cannot capture snapshot from top-level script.")
+
+    local_vars.append(get_filtered_global_variables(frame=frame))
```

Comment:
You're on the right track here, but you can simplify the code by merging this part into the while loop. Think about why you insert the check on line 43 in the first place, and what you can do now. (This change will also remove the need for the "Handling possible errors" part above.